hypertrace-core-graphql-span-schema/src/main/java/org/hypertrace/core/graphql/span/dao/SpanLogEventFetcher.java

```diff
@@ -189,4 +170,15 @@ public Object attribute(String key) {
       return this.attributeValues.get(key);
     }
   }
+
+  @lombok.Value
+  @Accessors(fluent = true)
+  private static class LogEventFilter implements FilterArgument {
+    FilterType type = FilterType.ATTRIBUTE;
```

Comment:
nit: we can remove the hardcoded attribute key in here and use an ID type filter (see: `EntityIdFilter`) if we map it in `HypertraceCoreAttributeScopeModule`. Eventually, those mappings will be moved into config. Likewise, the id on 148 can be looked up from that data (`AttributeStore.getForeignIdAttribute`)

Reply:
I tried adding it there, but the mapping is not present when this `AttributeStore.getForeignIdAttribute` call is made. Not sure if this mapping needs to be added somewhere else as well?

Reply:
Shouldn't need it anywhere else.
```java
    idBinder.addBinding().toInstance(forForeignId(LOG_EVENT, SPAN, "spanId"));
```

Maybe a flip flop of scopes either on the registration or request side? Could debug in and see what the id lookup table looks like at that point.

Reply:
Oh, later saw that this was already done :) looks right. One other thing that could be going wrong - is the attribute metadata already merged in whatever env you're testing in? The attribute store lookup first gets the key (that was registered as above), then uses that to fetch the attribute metadata from attribute service.

Reply:
It worked by adding it in `HypertraceAttributeScopeModule` (which seems to be overriding the binding in core)

Reply:
Ah crap, sorry. Completely forgot that the attribute scope modules are not additive (because enums can't inherit), each service has its own implementation.